### PR TITLE
ports/esp32/boards:add Waveshare ESP32-S3-Zero board.

### PR DESCRIPTION
**Pull request by siuwahzhong**
Waveshare ESP32-S3-Zero board use ESP32-S3FH4R2 chip, and board has a RGB led WS2812 on GPIO21

### CODE REVIEW

**Review by siuwahzhong — COMMENTED** [review]
ports/esp32/boards:add Waveshare ESP32-S3-Zero board

File: ports/esp32/boards/WAVESHARE_S3_ZERO/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+#endif
+#define MICROPY_HW_MCU_NAME                 "ESP32S3"
+
+#define MICROPY_PY_MACHINE_DAC              (0)
+
+// Enable UART REPL for modules that have an external USB-UART and don't use native USB.
+#define MICROPY_HW_ENABLE_UART_REPL         (1)
+
+#define MICROPY_HW_I2C0_SCL                 (9)
+#define MICROPY_HW_I2C0_SDA                 (8)
```
Comment (mattytrentini) [inline ports/esp32/boards/WAVESHARE_S3_ZERO/mpconfigboard.h:13]: Was there a reason to specify these? This board doesn't seem to explicitly define them for I2C.

File: ports/esp32/boards/WAVESHARE_S3_ZERO/mpconfigboard.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// Can be set by mpconfigboard.cmake.
+#define MICROPY_HW_BOARD_NAME               "ESP32-S3-Zero module"
+#endif
+#define MICROPY_HW_MCU_NAME                 "ESP32S3"
+
+#define MICROPY_PY_MACHINE_DAC              (0)
+
+// Enable UART REPL for modules that have an external USB-UART and don't use native USB.
+#define MICROPY_HW_ENABLE_UART_REPL         (1)
+
```
Comment (mattytrentini) [inline ports/esp32/boards/WAVESHARE_S3_ZERO/mpconfigboard.h:11]: The UART pins should probably be defined as they are explicitly labelled:

image

Unfortunately, I'm not sure which pins they're connected to (is there a schematic anywhere? I couldn't find one!).

**Review by mattytrentini — COMMENTED** [review]
Thanks for submitting, it will be good to have support for this handy little board! Some comments and queries inline.

File: ports/esp32/boards/WAVESHARE_S3_ZERO/mpconfigboard.h (hunk 2 above)
Comment (siuwahzhong) [reply ports/esp32/boards/WAVESHARE_S3_ZERO/mpconfigboard.h]: usb the UART0 default pin,TX0 is GPIO43, RX0 is GPIO44

**Pull request closed without merging**